### PR TITLE
[ENG-4488] Remove ember-cli-clipboard

### PR DESCRIPTION
**Pull request by brianjgeiger**
-   Ticket: [ENG-4488]
-   Feature flag: n/a

## Purpose

Ember-cli-clipboard was doing deprecated things, and upgrading it was a nightmare. The actual functionality is trivial to implement, though, so we've made our own version of its copy button.

## Summary of Changes

1. Add copy button
2. Add tests

## QA Notes

Test the copy button on the file detail page (hashes in the versions tab) and on the dev apps / tokens pages in all supported browsers, please.

[ENG-4488]: https://openscience.atlassian.net/browse/ENG-4488?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/osf-components/addon/components/copy-button/component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+interface Args {
+    success?: () => void;
+    error?: () => void;
+    disabled: Boolean;
+    clipboardText: string;
+}
+
+export default class CopyButton extends Component<Args> {
+    @task
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/copy-button/component.ts:13]: May not be needed now, but may want to pepper in a `@waitFor` here

File: lib/osf-components/addon/components/copy-button/component.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    async copy() {
+        assert('clipboardText is a required parameter.', Boolean(this.args.clipboardText));
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/copy-button/component.ts:15]: Yay asserts!

File: tests/integration/components/copy-button/component-test.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        clipboardContents = await navigator.clipboard.readText();
+        assert.equal(clipboardContents, 'Clip me');
+    });
+
+    test('It calls functions', async function(assert) {
+        assert.expect(2);
+
+        const callback = sinon.stub(navigator.clipboard, 'writeText');
+        callback.withArgs('Success');
+        callback.withArgs('Error').throws();
```
Comment (futa-ikeda) [inline tests/integration/components/copy-button/component-test.ts:52]: Neat!

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
Just one line suggestion for me. Love everything else!

**Review by adlius — APPROVED** [review]

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**